web/source/text/defaultOutput.ts

```diff
@@ -11,68 +11,118 @@ namespace com.keyman.text {
+        //   case Codes.keyCodes['K_TAB']:
+        //   case Codes.keyCodes['K_TABBACK']:
+        //   case Codes.keyCodes['K_TABFWD']:
+        //     return '\t';
+        //   default:
+        //     return '';
+        // }
+      }
+    }
+
+    public static isDOMCommand(Lkc: KeyEvent): boolean {
+      let code = DefaultOutput.codeForEvent(Lkc);
+
+      switch(code) {
+        case Codes.keyCodes['K_TAB']:
+        case Codes.keyCodes['K_TABBACK']:
+        case Codes.keyCodes['K_TABFWD']:
+          return true;
+        default:
+          return false;
+      }
+    }
+
+    /**
+     * forCommands - returns a boolean indicating if a non-text event should be triggered by the keystroke.
+     */
+    public static isCommand(Lkc: KeyEvent): boolean {
+      let code = DefaultOutput.codeForEvent(Lkc);
+
+      switch(code) {
+        // Should we ever implement them:
+        // case Codes.keyCodes['K_LEFT']:  // would not output text, but would alter the caret's position in the context.
+        // case Codes.keyCodes['K_RIGHT']:
+        //   return true;
+        default:
+          return this.isDOMCommand(Lkc);
+      }
+    }
+
     /**
      * This function is designed for future migration into a separate, explicitly-DOM-aware module.
      * It would then be assigned to this class via classic JS method extension practices as an 'override'
      * of `applyCommand`, reusing the base version of that name seen in this class, which is
      * designed for web-core.
      */
-    public static applyDOMCommand(outputTarget: OutputTarget, code: number, keyShiftState: number) {
-      let quiet = outputTarget instanceof Mock;
-
+    /**
+     * This function is designed for future migration into a separate, explicitly-DOM-aware module.
+     * It would then be assigned to this class via classic JS method extension practices as an 'override'
+     * of `applyCommand`, reusing the base version of that name seen in this class, which is
+     * designed for web-core.
+     * 
+     * for___Commands - returns boolean indicating if a non-text event should be triggered by the keystroke.
+     */
+    public static applyDOMCommand(Lkc: KeyEvent, keyShiftState: number) {
+      let code = DefaultOutput.codeForEvent(Lkc);
       let domManager = com.keyman.singleton.domManager;
 
       switch(code) {
         case Codes.keyCodes['K_TAB']:
-          if(!quiet) {
-            domManager.moveToNext(keyShiftState);
-          }
+          domManager.moveToNext(keyShiftState);
           break;
         case Codes.keyCodes['K_TABBACK']:
-          if(!quiet) {
-            domManager.moveToNext(true);
-          }
+          domManager.moveToNext(true);
           break;
         case Codes.keyCodes['K_TABFWD']:
-          if(!quiet) {
-            domManager.moveToNext(false);
-          }
+          domManager.moveToNext(false);
           break;
       }
     }
 
     public static applyCommand(Lkc: KeyEvent, keyShiftState: number) {
```

Comment:
Please add a function purpose comment and also following suggestion:

```suggestion
    public static applyCommand(Lkc: KeyEvent, keyShiftState: number): void {
```